server/plugin/plugin.go

```diff
@@ -96,3 +97,25 @@ func (p *Plugin) initBotUser() error {
 func (p *Plugin) ServeHTTP(c *plugin.Context, w http.ResponseWriter, r *http.Request) {
 	p.router.ServeHTTP(w, r)
 }
+
+func (p *Plugin) MessageWillBePosted(c *plugin.Context, post *model.Post) (*model.Post, string) {
+
+	// Check if message is a work item link.
+	if getTaskData, isValid := isValidTaskLink(post.Message); isValid {
+		post, msg := p.postTaskPreview(getTaskData, post.Message, post.UserId, post.ChannelId)
+		return post, msg
+	}
+	return nil, ""
+}
+
+// Function to validate the work item link.
+func isValidTaskLink(msg string) ([]string, bool) {
+	data := strings.Split(msg, "/")
+	if len(data) != 8 {
+		return nil, false
+	}
+	if (data[0] != constants.HTTPS && data[0] != constants.HTTP) || data[2] != constants.AzureDevopsBaseURL || data[5] != constants.Workitems || data[6] != constants.Edit {
```

Comment:
I guess this URL contains "edit" because you must be trying to edit the work item. What happens when you open this URL - 
 https://dev.azure.com/bs-test/bs-10/_workitems/186

Reply:
On clicking any workitem URL it consists of `edit`. On clicking the above URL we are redirected to `https://dev.azure.com/bs-test/bs-10/_workitems/assignedtome/`